src/api/Feed.js

```diff
@@ -0,0 +1,1022 @@
+class Feed extends Base {
+    /**
+     * @property {VersionsAPI}
+     */
+    versionsAPI: VersionsAPI;
+
+    /**
+     * @property {CommentsAPI}
+     */
+    commentsAPI: CommentsAPI;
+
+    /**
+     * @property {TasksAPI}
+     */
+    tasksAPI: TasksAPI;
+
+    /**
+     * @property {TaskAssignmentsAPI}
+     */
+    taskAssignmentsAPI: Array<TasksAPI | TaskAssignmentsAPI>;
+
+    /**
+     * @property {string}
+     */
+    id: string;
+
+    /**
+     * @property {boolean}
+     */
+    hasError: boolean;
+
+    constructor(options: Options) {
+        super(options);
+        this.taskAssignmentsAPI = [];
+    }
+
+    /**
+     * Creates a key for the cache
+     *
+     * @param {string} id folder id
+     * @return {string} key
+     */
+    getCacheKey(id: string): string {
+        return `feedItems_${id}`;
+    }
+
+    /**
+     * Gets the items from the cache
+     *
+     * @param {string} id the cache id
+     */
+    getCachedItems(id: string): ?FeedItemsCache {
+        const cache = this.getCache();
+        const cacheKey = this.getCacheKey(id);
+        return cache.get(cacheKey);
+    }
+
+    /**
+     * Sets the items in the cache
+     *
+     * @param {string} id - the cache id
+     * @param {Array} items - the feed items to cache
+     */
+    setCachedItems(id: string, items: FeedItems) {
+        const cache = this.getCache();
+        const cacheKey = this.getCacheKey(id);
+        cache.set(cacheKey, {
+            hasError: !!this.hasError,
+            items
+        });
+    }
+
+    /**
+     * Gets the feed items
+     *
+     * @param {BoxItem} file - The file to which the task is assigned
+     * @param {Function} successCallback - the success callback
+     * @param {Function} errorCallback - the error callback
+     */
+    feedItems(file: BoxItem, successCallback: Function, errorCallback: Function): void {
+        const { id } = file;
+
+        if (!id) {
+            throw getBadItemError();
+        }
+
+        const cachedItems = this.getCachedItems(id);
+        if (cachedItems) {
+            const { hasError, items } = cachedItems;
+            if (hasError) {
+                errorCallback(items);
+            } else {
+                successCallback(items);
+            }
+
+            return;
+        }
+
+        this.id = id;
+        this.hasError = false;
+        const versionsPromise = this.fetchVersions();
+        const commentsPromise = this.fetchComments();
+        const tasksPromise = this.fetchTasks();
+
+        Promise.all([versionsPromise, commentsPromise, tasksPromise]).then((feedItems) => {
+            const versions: ?FileVersions = feedItems[0];
+            const versionsWithRestoredVersion = this.addCurrentVersion(versions, file);
+            const unsortedFeedItems = [versionsWithRestoredVersion, ...feedItems.slice(1)];
+            const sortedFeedItems = this.sortFeedItems.apply(null, unsortedFeedItems);
+            this.setCachedItems(id, sortedFeedItems);
```

Comment:
Should this line above be guarded too in the if () block below?
Though maybe priming the cache for future might be useful.